pkg/storage/bucket/object_client_adapter.go

```diff
@@ -19,9 +24,36 @@ type ObjectClientAdapter struct {
 	isRetryableErr       func(err error) bool
 }
 
-func NewObjectClientAdapter(bucket, hedgedBucket objstore.Bucket, logger log.Logger, opts ...ClientOptions) *ObjectClientAdapter {
-	if hedgedBucket == nil {
-		hedgedBucket = bucket
+func NewObjectClient(ctx context.Context, backend string, cfg ConfigWithNamedStores, component string, hedgingCfg hedging.Config, logger log.Logger) (*ObjectClientAdapter, error) {
+	var (
+		storeType = backend
+		storeCfg  = cfg.Config
+	)
+
+	if st, ok := cfg.NamedStores.LookupStoreType(backend); ok {
+		storeType = st
+		// override config with values from named store config
+		if err := cfg.NamedStores.OverrideConfig(&storeCfg, backend, storeType); err != nil {
+			return nil, err
+		}
+	}
```

Comment:
I'm a bit confused by this part. We set `storeType` using `backend` but then also use `backend` to search for a namedStore.